src/plan/marksweep/metadata.rs

```diff
@@ -0,0 +1,85 @@
+use crate::util::constants;
+use crate::util::conversions;
+use crate::util::side_metadata::load_atomic;
+use crate::util::side_metadata::meta_bytes_per_chunk;
+use crate::util::side_metadata::store_atomic;
+use crate::util::side_metadata::try_mmap_metadata_chunk;
+use crate::util::side_metadata::SideMetadataScope;
+use crate::util::side_metadata::SideMetadataSpec;
+use crate::util::Address;
+use crate::util::ObjectReference;
+use conversions::chunk_align_down;
+use std::collections::HashSet;
+use std::sync::RwLock;
+
+lazy_static! {
+    pub static ref ACTIVE_CHUNKS: RwLock<HashSet<Address>> = RwLock::default();
```

Comment:
Is `ACTIVE_CHUNKS` still used? I thought you moved to side metadata. In this case, all code related to `ACTIVE_CHUNKS` can be removed.

Reply:
`ACTIVE_CHUNKS` is being used to check whether the metadata space for an address has already been mapped before the current metadata content for the address is read/written.
This is not needed for the rest of MMTk.

To remove this, side metadata should provide an additional method to perform this check.
We had this method in the side metadata before, but decided to remove it because:

1 - It is not required by the rest of MMTk.
2 - it can be implemented with `mmap(MAP_FIXED_NOREPLACE)` and `munmap()`, which is very expensive and requires assumptions on how the address space is being used, or
3 - it can be implemented with a pre-allocated metadata map over the metadata itself, which is again not needed for the rest of MMTk.

Reply:
I don't think keeping `ACTIVE_CHUNKS` is a good choice here, but I don't have a suitable fix which can be done in a reasonable time before the release.

Reply:
We should at least move it to the policy code, as it is per space. We should not create it as a global. 

`ACTIVE_CHUNKS` is not only for check whether metadata space is mapped. It is also used in the release phase. A MallocSpace will iterate through all the active chunks, do a linear scan through its address range for each chunk, check if there is an object for the address, then decide to free (if dead) or unset the mark bit (if alive). 

I am not sure if we care about performance for this `MallocSpace`. Using `ACTIVE_CHUNKS` to release malloc'd memory can be really bad for performance. A chunk will not be removed from `ACTIVE_CHUNKS` unless the whole chunk contains no live object. That means that we may have chunks with only a few survived objects, and we have to linearly scan through the entire chunk to find those objects.